packages/contracts-bedrock/src/L1/ETHLockbox.sol

```diff
@@ -165,9 +168,10 @@ contract ETHLockbox is ProxyAdminOwnedBase, Initializable, ISemver {
 
     /// @notice Authorizes a portal to lock and unlock ETH.
     /// @param _portal The address of the portal to authorize.
-    function _authorizePortal(address _portal) internal {
-        if (!_sameProxyAdminOwner(_portal)) revert ETHLockbox_DifferentProxyAdminOwner();
-        authorizedPortals[_portal] = true;
-        emit PortalAuthorized(_portal);
+    function _authorizePortal(IOptimismPortal _portal) internal {
+        if (!_sameProxyAdminOwner(address(_portal))) revert ETHLockbox_DifferentProxyAdminOwner();
+        if (_portal.superchainConfig() != superchainConfig) revert ETHLockbox_DifferentSuperchainConfig();
+        authorizedPortals[address(_portal)] = true;
+        emit PortalAuthorized(address(_portal));
```

Comment:
The addition of this casting suggests to me that we should just go all the way, and use the proper typing in the mapping and event.

Reply:
Good call, agreed

Reply:
Ok can fix